Orange/widgets/evaluate/owparameterfitter.py

```diff
@@ -318,93 +368,150 @@ def _add_plot(self):
+        layout.addWidget(edit, 4, 1)
+
+        # gui.lineEdit's connect does not call the callback on return pressed
+        # if the line hasn't changed.
+        @edit.returnPressed.connect
+        def _():
+            self.type = self.MANUAL
 
         gui.rubber(self.controlArea)
 
         gui.auto_apply(self.buttonsArea, self, "auto_commit")
 
+    def __on_type_changed(self):
+        self._settings_changed()
+
     def __on_parameter_changed(self):
         self._set_range_controls()
-        self.__on_setting_changed()
+        self._settings_changed()
+
+    def __on_min_max_changed(self):
+        self.type = self.FROM_RANGE
+        self._settings_changed()
 
-    def __on_setting_changed(self):
+    def __on_manual_changed(self):
+        self.type = self.MANUAL
+        self._settings_changed()
+
+    def _settings_changed(self):
         self._update_preview()
         self.commit.deferred()
 
     @property
     def fitted_parameters(self) -> list:
-        if not self._learner or not self._data:
+        if not self._learner:
             return []
-        return self._learner.fitted_parameters(self._data) \
-            if isinstance(self._learner, Fitter) \
-            else self._learner.fitted_parameters()
+        if isinstance(self._learner, Fitter):
+            if not self._data:
+                return []
+            return self._learner.fitted_parameters(self._data)
+        else:
+            return self._learner.fitted_parameters()
 
     @property
     def initial_parameters(self) -> dict:
-        if not self._learner or not self._data:
+        if not self._learner:
             return {}
-        return self._learner.get_params(self._data) \
-            if isinstance(self._learner, Fitter) else self._learner.params
+        if isinstance(self._learner, Fitter):
+            if not self._data:
+                return {}
+            return self._learner.get_params(self._data)
+        return self._learner.params
 
     @property
-    def steps(self) -> Iterable[int]:
+    def steps(self) -> tuple[int]:
+        self.Error.min_max_error.clear()
+        self.Error.manual_steps_error.clear()
         if self.type == self.FROM_RANGE:
-            step = 1
-            diff = self.maximum - self.minimum
-            if diff > 0:
-                exp = int(np.ceil(np.log10(diff + 1))) - 1
-                step = int(10 ** exp)
-            return range(self.minimum, self.maximum + step, step)
+            return self._steps_from_range()
         else:
-            try:
-                return [int(s) for s in self.manual_steps.split(",")]
-            except ValueError:
-                return []
+            return self._steps_from_manual()
+
+    def _steps_from_range(self) -> tuple[int]:
+        if self.maximum < self.minimum:
+            self.Error.min_max_error()
+            return ()
+
+        if self.minimum == self.maximum:
+            return (self.minimum, )
```

Comment:
I think this is reasonable, but we can remove it and extend the above condition from `<` to `<=`.